azext_iot/operations/events3/_builders.py

```diff
@@ -79,30 +79,36 @@ async def _evaluate_redirect(self, endpoint):
     async def _build_central_event_hub_target_async(self, cmd, app_id, central_api_uri):
         from azext_iot.common._azure import get_iot_central_tokens
 
-        tokens = get_iot_central_tokens(cmd, app_id, central_api_uri)
-        eventHubToken = tokens['eventhubSasToken']
-        hostnameWithoutPrefix = eventHubToken['hostname'].split("/")[2]
-        endpoint = hostnameWithoutPrefix
-        path = eventHubToken["entityPath"]
-        tokenExpiry = tokens['expiry']
-        auth = self._build_auth_container_from_token(endpoint, path, eventHubToken['sasToken'], tokenExpiry)
-        address = "amqps://{}/{}/$management".format(hostnameWithoutPrefix, eventHubToken["entityPath"])
-        meta_data = await self._query_meta_data(address, path, auth)
-        partition_count = meta_data[b'partition_count']
-        partition_ids = []
-        for i in range(int(partition_count)):
-            partition_ids.append(str(i))
-        partitions = partition_ids
-        auth = self._build_auth_container_from_token(endpoint, path, eventHubToken['sasToken'], tokenExpiry)
+        allTokens = get_iot_central_tokens(cmd, app_id, central_api_uri)
+        targets = []
+        # create target for each event hub
+        for key in allTokens:
+            tokens = allTokens[key]
```

Comment:
maybe worth extracting a function here?

```
for _, tokens in allTokens:
  event_hub_target = self._build_event_hub_target(tokens)
  targets.append(event_hub_target)
```

Which could be made into a list comprehension if you really wanted to

targets = [create_target(tokens) for key, tokens in allTokens.items()]